### DISCUSSION TITLE
VS Code debugger's `launch.json` configuration

### DISCUSSION THREAD

**Discussion by InSuperposition** [solidjs/solid-start#1567] [General]
I am attempting to setup debugging of a solid-start bare starter app but not having much luck with either `attach` or `launch` stopping at breakpoints. I've looked over Nitro and Vinxi's issues/discussions for a similar topic but have not had much luck.
Does anyone have suggestions on setting up the debug configuration for node/browser in VS Code.

[VS Code debugger options](https://github.com/microsoft/vscode-js-debug/blob/main/OPTIONS.md)

**Comment by remy90** [comment]
I've also seen this question appear at least twice in the Discord server. Having documentation of how SolidStart can connect to VSCode's debugger is likely a must-have for many

**Comment by A2va** [comment]
I have this configuration:
```json
{
    "configurations": [
        {
            "name": "Firefox Debug",
            "type": "firefox",
            "request": "launch",
            "reAttach": true,
            "url": "http://localhost:3000/",
            "webRoot": "${workspaceFolder}/src",
            "pathMappings": [
                {
                    "url": "http://localhost:3000/_build/@fs/D:/GitHub/a2va.dev/src",
                    "path": "${workspaceFolder}/src" 
                },
                {
                    "url": "http://localhost:3000/_build/src",
                    "path": "${workspaceFolder}/src"
                },
                {
                    "url": "http://localhost:3000/_build/",
                    "path": "${workspaceFolder}/src"
                }
            ]
        },
        {
            "type": "msedge",
            "request": "launch",
            "name": "Edge Debug",
            "url": "http://localhost:3000",
            "webRoot": "${workspaceFolder}",
            "sourceMaps": true,
            "pathMapping": {
                "_build/@fs/D:/GitHub/a2va.dev/src": "${workspaceFolder}/src",
                "_build/src/": "${workspaceFolder}/src",
                "_build/": "${workspaceFolder}/src"
            }
        }
    ]
}
````
The only problem I have is that one of the urls contain the path to the project directory, I tried to use `${workspaceFolder}` but it didn't work either because `${workspaceFolder}` is not supported in pathMappings or because the variable is replaced by `D:\GitHub\a2va.dev\src` it doesn't work.

You also need to enable the sourcemap in the vite configuration.

**Comment by paulo-assoc** [comment]
Did anyone figure out how to set up the debug configuration for node/browser in VS Code?

**Comment by bruceharrison1984** [comment]
This works with SolidStart server-side code right OOTB
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Launch Solid-App",
      "request": "launch",
      "runtimeExecutable": "npm",
      "runtimeArgs": ["run", "dev"],
      "cwd": "${workspaceFolder}/src",
      "type": "node"
    }
  ]
}
```

just run the task, then open your browser to the correct url:port. Debug points should activate once the page loads.
